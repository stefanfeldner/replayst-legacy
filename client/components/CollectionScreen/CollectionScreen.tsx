import { useContext } from 'react';
import { View, StyleSheet, Text, ViewStyle } from 'react-native';
import GameList from '../GameList/GameList';
import { UserContext } from '../UserContext/UserContext';
import { Game } from '../../types/Game';

export default function CollectionScreen() {
  const { toRender }: { toRender: [Game[], Function], } = useContext(UserContext);
  const [tiles] = toRender;
  
  return (
    <View style={styles.container}>
      {!tiles ? (
        <Text>Loading...</Text>
      ) : (
        <GameList tiles={tiles} isFromCollection={true} />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#110d07',
  },
});
