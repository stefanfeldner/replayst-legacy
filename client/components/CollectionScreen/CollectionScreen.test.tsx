import CollectionScreen from './CollectionScreen';
import { render } from '@testing-library/react-native';

test('should render correctly', () => {
  const tree = render(<CollectionScreen />);
  expect(tree).toMatchSnapshot();
});