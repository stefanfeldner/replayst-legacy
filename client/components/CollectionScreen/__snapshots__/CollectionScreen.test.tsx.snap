// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#110d07",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      extraData={18}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={4}
      keyExtractor={[Function]}
      maxToRenderPerBatch={3}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-around",
                "marginTop": 20,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 10,
                      "borderBottomRightRadius": 10,
                      "borderTopLeftRadius": 10,
                      "borderTopRightRadius": 10,
                      "height": 35,
                      "justifyContent": "center",
                      "width": 110,
                    },
                    Object {
                      "backgroundColor": "#20150d",
                    },
                  ]
                }
              >
                <Text />
              </View>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 10,
                      "borderBottomRightRadius": 10,
                      "borderTopLeftRadius": 10,
                      "borderTopRightRadius": 10,
                      "height": 35,
                      "justifyContent": "center",
                      "width": 110,
                    },
                    Object {
                      "backgroundColor": "#20150d",
                    },
                  ]
                }
              >
                <Text />
              </View>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderBottomLeftRadius": 10,
                      "borderBottomRightRadius": 10,
                      "borderTopLeftRadius": 10,
                      "borderTopRightRadius": 10,
                      "height": 35,
                      "justifyContent": "center",
                      "width": 110,
                    },
                    Object {
                      "backgroundColor": "#20150d",
                    },
                  ]
                }
              >
                <Text />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "alignContent": "center",
          "backgroundColor": "#20150d",
          "borderBottomLeftRadius": 20,
          "borderBottomRightRadius": 20,
          "borderTopLeftRadius": 20,
          "borderTopRightRadius": 20,
          "bottom": 24,
          "justifyContent": "space-between",
          "padding": 5,
          "paddingVertical": 12,
          "position": "absolute",
          "right": 22,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text />
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
